### DISCUSSION TITLE
Assembleia + Votação online

### DISCUSSION THREAD

**Discussion by mazulo** [apyb/comunidade#6] [Geral]
No dia 14 de novembro de 2021, nós vamos ter a [Assembleia 2021](https://python.org.br/assembleia2021/). Na mesma teremos além da usual apresentação da situação atual da Associação Python Brasil, a votação para nova diretoria. Toda a assembleia acontecerá de maneira online, e da mesma maneira, a votação. Gostaria aqui de abrir a discussão de um modelo sobre como esse processo será feito. Será feito no boca a boca durante a chamada? Poderíamos utilizar alguma ferramenta open source? Ou quem sabe criar um sistema interno ao associados de montagem de chapa e votação?

Caso a última seja escolhida, a decisão daqui prosseguirá para a issue abaixo. Caso contrário, será fechada.

> A issue abaixo foi convertida em uma discussion, link abaixo

- [x] https://github.com/pythonbrasil/associados/issues/221
- [ ] https://github.com/pythonbrasil/associados/discussions/223

**Comment by lucianoratamero** [comment]
a única coisa que consegui pensar para a parte da votação foi que, caso estejamos usando algo como o Discord, poderíamos usar reações como contagem de votos, apesar de isso ter vários problemas.

se necessário, acho que podemos fazer uma plataforma para votação, de repente até mesmo no site da apyb. temos algumas semanas até lá, e não acho que deva ser muito difícil, e isso tambṕem deixaria o código da votação auditável.

**Reply by mazulo** [reply]
@lucianoratamero  concordo com a parte disso ficar dentro do site da apyb:
- isso deixaria o código da votação auditável, como você comentou
- fica **muito** mais fácil de colocar controles (como quantidade de votos por pessoa e etc)
- é algo que vai poder ficar para os próximos anos, tanto para registro de chapas quanto para algum cenário onde a votação precise ser online novamente

Me coloco a disposição para implementarmos isso

**Reply by rougeth** [reply]
Durante a organização da pybr2021 tivemos que fazer algumas votações e encontramos um fluxo que funcionou várias vezes usando emojis:
```
Voto na sua chapa:
🍊 - Chapa 1
🥑 - Chapa 2
🥥 - Chapa 3
```
Daí o emoji que tinha mais votação, era a opção escolhida. Super simples. Um bot ajudaria na hora de listar quem clicou em qual emoji.

Único ponto que vem depois é como garantir que quem votou, pode de fato votar.

**Reply by lucianoratamero** [reply]
também tem o ponto de que as pessoas, nesse caso, poderiam votar em mais de uma pessoa, o que talvez não seja desejável =/

**Reply by rennerocha** [reply]
@rougeth acho que esse tipo de votação funciona bem para coisas não "sérias". Eu tenho minhas dúvidas se é uma maneira válida legalmente para a eleição de uma diretoria de associação.

Não podemos esquecer que os resultados dessa votação precisam ser válidas para registro. Caso alguma parte (por exemplo uma chapa perdedora) questione, pode dar algum tipo de dor de cabeça por ser um sistema que é facilmente burlável e de difícil controle de quem poderia ou não votar.

Como temos menos de um mês para a Assembleia, se não pagarmos ninguém, não vai sair nenhum sistema de votação. Talvez valesse a pena verificar que tipo de sistema condomínios por exemplo estão utilizando para fazer suas votações em suas assembleias (o meu por exemplo vai ser hibrido na proxima assembleia) e analisar a possibilidade de contratar um desses sistemas para essa assembleia em específico.

**Reply by gabubellon** [reply]
@rennerocha  vou colocar alguns pontos:

## Sobre o bot no Discord
*  No evento desse anos usamos um bot para abrir inscrições nos tutorias, podemos usar a mesma lógica;
*  Dentro de um cenário de votação podemos enviar uma mensagem com a chapa:
```
CHAPA 1
CHAPA 2
CHAPA 3

Envie !votar CHAPA N para votar na sua chapa
```
* O bot consegue pegar o id/nick de quem votou e:
   1. Logar em um canal (o mesmo ou outro de votação) a ocorrência de votação;
   2. Logar em um arquivo físicos no servidor que o bot roda o mesmo item
* Esse log poder ter "Pessoas VOTOU" ou até "PESSOA VOTOU EM CHAPA"
* É possível realizar validações em memória e no arquivado auditável de  QUEM JÁ VOTOU !!
* Com isso eu acho que a gente garante:
  1. Auditoria interna e externa (dentro do Discord e pegamos o arquivo do servidor)
  2. Temos um interação dentro do Discord
  3. O Código do Bot é aberto e auditável
  
Problemas desse item:
* Usuários do discord com nick name Th3ki1ler = quem é esse ser na associação (solução, OBRIGAR as pessoas a usar um nick name auditável dentro do server ou pedir o EMAIL do associado na votação !votarchapa EMAIL CHAPA
* Liberar votação APENAS para associados (como saber que o usar do discord é um associados ??) . Solução FORM para associados pedindo o ID (e ensinado a pegar o ID) do Discord
* Não é um problema, mas podemos pensar em votar em pvt no bot e não no canal

[Aqui](https://github.com/rougeth/pybr2021-discord-bot/blob/db4fdbf21c7cc9b8c64b0927f50e531f85f05a35/pybr2021/cogs/tutorial.py#L307) um exemplo de como pegar as mensagens para o bot (nesse caso tá faltando um semaforo dos async)
Mas isso permite 
* Ter um código de votação ainda que para o Discord Aberto e auditável
* Ter um arquivo externo ao discord auditável para ser disponibilzado a comunidade
* Temos pessoas (eu e @rougeth e qualquer pessoa que saida async) capaz de criar esse bot em 1 final de semana (o tutorias eu fiz em 3hs!!)

Exemplo de log no Discord após usar o comando `entrar` em um tutorial
![image](https://user-images.githubusercontent.com/7385097/138460498-95257f7f-d755-4294-8b0b-83f0bf43327c.png)



Ainda  existem outras soluções prontas no discord e com código aberto
https://www.google.com/amp/s/pythonawesome.com/discord-bot-for-polls-and-votes-including-stv/amp/
https://eryn.io/Votum/

Temos a opção de usar uma feraamenta de mercado que já tenha isso
https://support.zoom.us/hc/en-us/articles/213756303-Polling-for-meetings?mobile_site=true#h_ee8443c8-e8ac-448c-891f-fb4400e45d20
O zoom permite exportar os pools realidados em reuniões, em outra organização que participo usamos o mesmo, é pago, as acho que custo não é um problema dado o tempo
